src/main/java/org/apache/mesos/hdfs/Scheduler.java

```diff
@@ -256,6 +257,12 @@ private boolean launchNode(SchedulerDriver driver, Offer offer,
     //executorName is to distinguish different types of nodes
     //taskType is the type of task in mesos to launch on the node
     //taskName is a name chosen to identify the task in mesos and mesos-dns (if used)
```

Comment:
Yeah, I don't see how this change helps/hurts the code, so I'd rather not change it unnecessarily.
>  Ideally, we should never be calling launchTask if we don't want to launch a task.
+1, sounds better than finding out in the middle of the call that you don't "need" to launch

I'd also say that if we were to pull getNextTaskName up to the tryToLaunchX callers, then maybe break it into getNextJNName/getNextNNName. Also seems excessive we're sometimes passing the taskType to launchTask 3 times: as nodeName, taskType, and taskName (all the same for DN)